Source/WebCore/animation/WebAnimationUtilities.cpp

```diff
@@ -335,34 +333,29 @@ String pseudoIdAsString(PseudoId pseudoId)
     case PseudoId::ViewTransition:
         return viewTransition;
     case PseudoId::ViewTransitionGroup:
-        return viewTransitionGroup;
+        return makeString("::view-transition-group"_s, '(', pseudoElementIdentifier->nameArgument, ')');
     case PseudoId::ViewTransitionImagePair:
-        return viewTransitionImagePair;
+        return makeString("::view-transition-image-pair"_s, '(', pseudoElementIdentifier->nameArgument, ')');
     case PseudoId::ViewTransitionOld:
-        return viewTransitionOld;
+        return makeString("::view-transition-old"_s, '(', pseudoElementIdentifier->nameArgument, ')');
     case PseudoId::ViewTransitionNew:
-        return viewTransitionNew;
+        return makeString("::view-transition-new"_s, '(', pseudoElementIdentifier->nameArgument, ')');
     case PseudoId::WebKitScrollbar:
         return webkitScrollbar;
     default:
         return emptyString();
     }
 }
 
-std::optional<PseudoId> pseudoIdFromString(const String& pseudoElement)
+std::pair<bool, std::optional<Style::PseudoElementIdentifier>> pseudoElementIdentifierFromString(const String& pseudoElement, Document* document)
```

Comment:
It's really unclear (to me) what the difference between the `optional` being std::nullopt, and `PseudoId::None`. Even the bool is pretty non-obvious, except when looking at callers that provide a variable name that shows it's for parse failures.

I realise that isn't new to this patch, and you're just expanding usage, but it'd be nice for a comment to briefly cover the various 'no' states.